src/main/java/com/google/sps/travelbud/City.java

```diff
@@ -74,6 +74,46 @@ public static City getCity(DatastoreService datastore, long id) {
     }
   }
 
+  public static List<City> filterName(DatastoreService datastore, String name) {
+    List<City> filteredCities = new ArrayList<>();
+    List<City> cities = City.getAll(datastore);
+    for (City C : cities) {
+      String nameOfC = C.getName().toLowerCase();
+      String substr = nameOfC.substring(0, name.length());
+      if (nameOfC.length() >= name.length() && substr.equals(name.toLowerCase())) {
+        filteredCities.add(C);
+      }
+    }
+    return filteredCities;
```

Comment:
An easier way to write this would be:
```java
return City.getAll(datastore).stream().filter(c -> c.getName().toLowerCase().contains(name.toLowerCase())).collect(Collectors.toList());
```